app/helpers/posts_helper.rb

```diff
@@ -0,0 +1,11 @@
+module PostsHelper
+
+  private 
+
+  def logged_in_user
+    unless logged_in? 
+      flash[:danger] = "You must log in in order to submit a post."
+      redirect_to login_url 
+    end 
+  end 
```

Comment:
Thanks for the clarification - I certainly was not being mindful of where they ought to live and their purpose. The `logged_in?` and `current_user` functions are called in the `_header.html.erb` partial view, so I kept those in the SessionsHelper file, because moving them all into ApplicationController was not giving that view access to those functions. Let me know if I was simply missing something, but the way it is now, I've included the logging methods within `ApplicationController` because they are utilized by sub-controllers, while keeping the boolean `logged_in?` and `current_user` methods within SessionsHelper, because they are utilized in a view (I simply deleted the `PostsHelper` file because it was clearly unnecessary).